--- original
+++ remediated
@@ -0,0 +0,0 @@
 package com.skf.labs.sqli;
 
@@ -3,0 +3,1 @@
+import org.springframework.jdbc.core.RowMapper;
 import java.util.List;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
 import org.springframework.jdbc.core.JdbcTemplate;
 
 
 @Component
 public class SqliModel {
 
 @Autowired
 private JdbcTemplate jdbcTemplate;
 
@@ -16,3 +17,2 @@
-public List<Page> getPage(String pageId){
-    String sql = "SELECT pageId, title, content FROM pages WHERE pageId="+pageId;
-    List<Page> pages = jdbcTemplate.query(sql, (resultSet, rowNum) -> new Page(resultSet.getInt("pageId"),resultSet.getString("title"), resultSet.getString("content")));
+    String sql = "SELECT pageId, title, content FROM pages WHERE pageId=?";
+    List<Page> pages = jdbcTemplate.query(sql, new Object[]{pageId}, (resultSet, rowNum) -> new Page(resultSet.getInt("pageId"),resultSet.getString("title"), resultSet.getString("content")));
     return pages;
 }
     
 }